examples/02_first_level_models/plot_predictions_residuals.py

```diff
@@ -0,0 +1,121 @@
+
+
+#########################################################################
+# Fit model
+# ---------
+# Note that `minimize_memory` is set to `False` so that `FirstLevelModel`
+# stores the residuals.
+# `signal_scaling` is set to False, so we keep the same scaling as the
+# original data in `fmri_img`.
+
+fmri_glm = FirstLevelModel(t_r=7,
+                           drift_model='cosine',
+                           signal_scaling=False,
+                           mask_img=mask,
+                           minimize_memory=False)
+
+fmri_glm = fmri_glm.fit(fmri_img, events)
+
+
+#########################################################################
+# Calculate and plot contrast
+# ---------------------------
+z_map = fmri_glm.compute_contrast('active - rest')
+
+plotting.plot_stat_map(z_map, bg_img=mean_img, threshold=3.1)
+
+#########################################################################
+# Extract the largest clusters
+# ----------------------------
+
+table = get_clusters_table(z_map, stat_threshold=3.1,
+        cluster_threshold=20).set_index('Cluster ID', drop=True)
+table.head()
+
+masker = input_data.NiftiSpheresMasker(table.loc[range(1,7), ['X', 'Y', 'Z']].values)
+
+real_timeseries = masker.fit_transform(fmri_img)
+predicted_timeseries = masker.fit_transform(fmri_glm.predicted)
+
+
+#########################################################################
+# Plot predicted and actual time series for 6 most significant clusters
+# ---------------------------------------------------------------------
+for i in range(1, 7):
+    plt.subplot(2,  3, i)
+    plt.title('Cluster peak {}\n'.format(table.loc[i, ['X', 'Y', 'Z']].tolist()))
+    plt.plot(real_timeseries[:, i-1], c='k', lw=2)
+    plt.plot(predicted_timeseries[:, i-1], c='r',  ls='--', lw=2)
+    plt.xlabel('Time')
+    plt.ylabel('Signal intensity')
+    
+plt.gcf().set_size_inches(12, 7)
+plt.tight_layout()
+
+#########################################################################
+# Get residuals
+# -------------
+resid = masker.fit_transform(fmri_glm.residuals)
+
+
+#########################################################################
+# Plot distribution of residuals
+# ------------------------------
+# Note that residuals are not really distributed normally.
+
+
+for i in range(1, 7):
+    plt.subplot(2,  3, i)
+    plt.title('Cluster peak {}\n'.format(table.loc[i, ['X', 'Y', 'Z']].tolist()))
+    plt.hist(resid[:, i-1])
+    print('Mean residuals: {}'.format(resid[:, i-1].mean()))
+
+plt.gcf().set_size_inches(12, 7)
+plt.tight_layout()
+
+
+#########################################################################
+# Plot R-squared
+# --------------
+# Because we stored the residuals, we can plot the R-squared: the proportion
+# of explained variance of the GLM as a whole. Note that the R-squared is markedly
+# lower deep down the brain, where there is more physiological noise and we 
+# are further away from the receive coils.
```

Comment:
I'm not a big fan of R squared in this context: as it is calculated in-sample, it has no mechanism to rpevent overfit, hence a bigger model always has a higher Rsquared.
Besides, this one does not make the distinction between effects of interests (conditions) and effects of no interest (drifts, motion), so the interpretation is not straightforward.
This probably deserves a comment, and possibly a plot of the "effects of interest" F test.
Does  this make sense ?

Reply:
@bthirion has this been resolved to your satisfaction?

Reply:
I added more text to illustrate the drawbacks of r-squared and added a section for calculating/plotting an f-test.